### PR TITLE
feat: support google.golang.org/protobuf

### PR DESCRIPTION
**Pull request by aaronc**
This adds support for:
* marshaling/unmarshaling `google.golang.org/protobuf/proto.Message` types
* jsonpb marshaling/unmarshaling these types
* getting the message name for these types using `MessageName`

This should remove any incompatibilties or need for type switching the SDK has when using these types.

Given that gogo proto already does a bunch of type switches to see what kind of message is used, this support seems reasonable and goes a long way to providing v2 support for anyone who wants it (we will use this at Regen at least and others will have a choice). Basically, this makes v2 types more with the v1 APIs the SDK uses without needing to change anything on the SDK side.

### CODE REVIEW

File: proto/table_marshal.go [hunk 1]
```diff
@@ -2944,6 +2946,9 @@ func Marshal(pb Message) ([]byte, error) {
 		// NOTE: This is not efficient.
 		return m.Marshal()
 	}
+	if m, ok := pb.(protov2.Message); ok {
```
Comment (tac0turtle) [inline proto/table_marshal.go:2949]: should use 

var marshalOption = proto.MarshalOptions{
	Deterministic: true,
}
to be safe and consistent

File: jsonpb/jsonpb.go [hunk 2]
```diff
@@ -128,6 +131,19 @@ type JSONPBUnmarshaler interface {
 
 // Marshal marshals a protocol buffer into JSON.
 func (m *Marshaler) Marshal(out io.Writer, pb proto.Message) error {
+	if msgV2, ok := pb.(protov2.Message); ok {
```
Comment (kocubinski) [inline jsonpb/jsonpb.go:134]: what is the thinking behind placing this v2 code path (and early return) before the nil and required field checks below which the v1 code path follows?

Comment (aaronc) [reply jsonpb/jsonpb.go:134]: That stuff except for maybe the nil check would be invalid for v2. Also this is a jsonpb so not performance critical.

**Review by kocubinski — APPROVED** [review]

**Review by julienrbrt — COMMENTED** [review]
can we get a changelog?

**Comment by aaronc** [comment]
> can we get a changelog?

Added

**Review by julienrbrt — APPROVED** [review]
lgtm! We should alias MsgTypeUrl from core and codec in the SDK to this (+ /) 🙏

**Pull request merged**